File: ios/Sources/EyenetNetworkInfoPlugin/EyenetNetworkInfoPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(EyenetNetworkInfoPlugin)
public class EyenetNetworkInfoPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "EyenetNetworkInfoPlugin"
    public let jsName = "EyenetNetworkInfo"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = EyenetNetworkInfo()

    @objc func echo(_ call: CAPPluginCall) {
       
        call.resolve([
            "network": "example"
        ])
    }
}

File: android/src/main/java/com/eyenet/mobile/network/info/EyenetNetworkInfoPlugin.java
package com.eyenet.mobile.network.info;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "EyenetNetworkInfo")
public class EyenetNetworkInfoPlugin extends Plugin {

    private EyenetNetworkInfo implementation = new EyenetNetworkInfo();

    @PluginMethod
    public void echo(PluginCall call) {
        ret.put("network", implementation.getNetworkInfo());
        call.resolve(ret);
    }
}
